cpio/test/test_extract_cpio_absolute_paths.c

```diff
@@ -0,0 +1,67 @@
+/*-
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright (c) 2024 Mostyn Bramley-Moore <mostyn@antipode.se>
+ */
+
+#include "test.h"
+
+#include <stdlib.h>
+
+#if defined(_WIN32) && !defined(__CYGWIN__)
+
+#include <fileapi.h>
+
+#define UNLINK _unlink
+
+#else
+
+#define UNLINK unlink
+
+#endif
+
+DEFINE_TEST(test_extract_cpio_absolute_paths)
+{
+	int r;
+
+#if defined(_WIN32) && !defined(__CYGWIN__)
+	char temp_dir[MAX_PATH + 1];
+	assert(GetTempPathA(MAX_PATH + 1, temp_dir) != 0);
+
+	char temp_file_name[MAX_PATH + 1];
+	char temp_absolute_file_name[MAX_PATH + 1];
+	assert(GetTempFileNameA(temp_dir, "abs", 0, temp_file_name) != 0);
+
+	assert(_fullpath(temp_absolute_file_name, temp_file_name, MAX_PATH) != NULL);
+#else
+	char temp_absolute_file_name[] = "/tmp/cpio-noabs.testXXXXXX";
```

Comment:
Another way to build an absolute file name here is to base it on `testworkdir`.  This is a global variable defined in test.h which holds the full pathname of the current working directory.  That directory was created specifically for this test to use, it is is guaranteed to be empty when the test starts and will be automatically cleaned up by the test runner after this test ends.

You can take advantage of that in the Windows variant by deleting your `temp_dir` variable above and using `testworkdir` as the base path for the `GetTempFileNameA` call.  For POSIX, just append `/somefilename` to `testworkdir`.